kore/src/Kore/Repl/Data.hs

```diff
@@ -39,6 +40,14 @@ import           Kore.Step.Simplification.Data
                  ( Simplifier )
 import qualified Kore.Step.Strategy as Strategy
 
+newtype AxiomIndex = AxiomIndex
```

Comment:
:+1:

Reply:
Plan is to replace the other `Int`s with the respective `*Index`. Probably add `NodeIndex` too.